vbfml/scripts/err_analysis

```diff
@@ -455,5 +489,313 @@ def compare_hist(ctx, input_dirs: str, number_epochs: int, save_name: str) -> No
     plt.savefig(pjoin(outdir, f"comparison_val_loss_{save_name}.pdf"))
 
 
+@cli.command()
+@click.pass_context
+@click.option(
+    "-i",
+    "--input-dir",
+    required=True,
+    help="path to the prediction directory of a model",
+)
+def roc(ctx, input_dir: str) -> None:
+    """
+    compute the roc curve of the model. /!/ input_dir point at the prediction_dir without a '/' at the end.
+    only compute and write in pkl file so it allows to use plot-roc on several model to compare in in one plot
+    """
+
+    # load model and prediction pickle file
+    loader = TrainingLoader(os.path.dirname(input_dir))
+    model = loader.get_model()
+    features, predictions, label_encoding, labels = load_cache(input_dir)
+    mjj = features["mjj"]
+    detajj = features["detajj"]
+
+    # compute ROC and precision recall curve
+    precision, recall, thresholds_prc = precision_recall_curve(labels, predictions)
+    fpr, tpr, thresholds_roc = roc_curve(labels, predictions)
+
+    # do the same for a simple cut on mjj and detajj
+    mjj_norm = mjj / mjj.max()
+    detajj_norm = detajj / detajj.max()
+    fpr_m, tpr_m, thresh_roc_m = roc_curve(labels, mjj_norm)
+    fpr_eta, tpr_eta, thresh_roc_eta = roc_curve(labels, detajj_norm)
+
+    # write data into pickle file
+    print("write data into pickle file")
+    data = {
+        "fpr": fpr,
+        "tpr": tpr,
+        "thresh_roc": thresholds_roc,
+        "precision": precision,
+        "recall": recall,
+        "thresholds_prc": thresholds_prc,
+        "fpr_m": fpr_m,
+        "tpr_m": tpr_m,
+        "mjj_max": mjj.max(),
+        "thresh_roc_m": thresh_roc_m,
+        "fpr_eta": fpr_eta,
+        "tpr_eta": tpr_eta,
+        "thresh_roc_eta": thresh_roc_eta,
+        "eta_max": detajj.max(),
+    }
+
+    with open(f"{input_dir}/roc.pkl", "wb") as f:
+        pickle.dump(data, f)
+
+
+def histogram(
```

Comment:
A few things about this function:

- This is not called directly from the command line right? So I think this can be moved to another file, I'm thinking maybe `plot/util.py`? So that under `err_analysis` file we only keep what we call directly, and we can import this function for better code refactoring.

- Is this to plot any generic histogram, given a pandas DataFrame? It might be better to rename it to something like `plot_histogram`? Or if you are plotting two labels side-by-side, it could be something like `plot_histograms_for_each_label`, so that the name of the function hints at what it's doing.

- Also, could you add a brief docstring, just describing what this function is doing?

Reply:
no this function is quite specific to plot histogram of feature according to their true label, 
I just implemented it because at the beginning I was plotting three time the exact same kind of histogram and thought it would cleaner like this

Reply:
I made the corrections

Reply:
Thanks for clarifying, yeah it is always a good idea to refactor re-used code like that I think:) Thanks for making the changes though, I think the new name of the function is more descriptive of what it's doing, and I think it's nice to have it in `plot/util.py` in case we want to use it somewhere else too. Let me know if you think it looks good now, or you want some changes to this.